src/cloudant/document.py

```diff
@@ -356,6 +360,7 @@ def __setitem__(self, key, value):
         if key == '_id':
             self._document_id = value
         super(Document, self).__setitem__(key, value)
+        self.dirty = True
```

Comment:
What if a user has edited the doc using `.update(d2)`, `.pop(key)`, `.clear()` etc?

Reply:
I fear we might need to store a hash of the contents and compare each time. Or override all these methods too ofc.

Reply:
I'm currently working on overriding MutableMapping methods and also observe static methods stuff, I hope I can push something later today :)